core/src/worker/client.rs

```diff
@@ -142,8 +150,8 @@ pub(crate) trait WorkerClient: Sync + Send {
         query_result: QueryResult,
     ) -> Result<RespondQueryTaskCompletedResponse>;
 
-    #[allow(clippy::needless_lifetimes)] // Clippy is wrong here
-    fn capabilities<'a>(&'a self) -> Option<&'a get_system_info_response::Capabilities>;
+    fn replace_client(&self, new_client: RetryClient<Client>);
+    fn capabilities(&self) -> Arc<Option<get_system_info_response::Capabilities>>;
```

Comment:
This should stay the same. It's generally nicer to return a ref to callers since they can do the clone if they need to, and if they don't the clone is avoided. Esp after getting rid of the Arc

Reply:
Will try, I forget the issues I had here keeping the ref. But my whole goal with the Arc was to avoid clone/copy. But in the other comment it sounds like we want clone/copy.

Reply:
Commented above, this was to avoid `WorkerClient::capabilities()` call from having to copy the capabilities on every invocation (can't be a reference anymore since client is replaced and lifetime no longer tied to self).

Reply:
Changed this to clone on every call now